eventuals/task.h

```diff
@@ -166,18 +167,59 @@ struct _TaskFromToWith {
     Fail = 2,
   };
 
-  template <typename K_, typename From_, typename To_, typename... Args_>
+  // Needed as an alternative for 'std::monostate' to avoid compile errors
+  // due to the same function prototypes.
+  struct EmptyStruct {};
+
+  template <
+      typename K_,
+      typename From_,
+      typename To_,
+      typename... Args_>
   struct Continuation {
+    using Dispatch_ =
+        Callback<
+            Action,
+            std::optional<std::exception_ptr>&&,
+            Args_&&...,
+            // Can't have a 'void' argument type
+            // so we are using 'std::monostate'.
+            std::optional<
+                std::conditional_t<
+                    std::is_void_v<From_>,
+                    std::monostate,
+                    From_>>&&,
+            std::unique_ptr<void, Callback<void*>>&,
+            Interrupt&,
+            std::conditional_t<
+                std::is_void_v<To_>,
+                Callback<>&&,
+                Callback<To_>&&>,
+            Callback<std::exception_ptr>&&,
+            Callback<>&&>;
+
     template <typename... From>
     void Start(From&&... from) {
-      if constexpr (std::is_void_v<From_>) {
-        Dispatch(Action::Start, std::monostate{});
+      if (dispatch_.index() == 0) {
```

Comment:
Please use a switch statement on dispatch_.index().

It would help to clarify that the else case is when index == 2.